R/csv_to_sqlite.R

```diff
@@ -0,0 +1,92 @@
+#' in the same sqlite table database before the next chunk is loaded into
+#' memory.
+#' @param csv_file name of the text file to convert
+#' @param sqlite_file name of the newly created sqlite file
+#' @param table_name name of the table to store the data table in the sqlite
+#'   dbase
+#' @param delim text file delimiter (default ",")
+#' @param pre_process_size the number of lines to check the data types of the
+#'   individual columns (default 1000)
+#' @param chunk_size the number of lines to read for each chunk (default 50000)
+#' @param quote same as in \code{readr::read_delim} (default "\"")
+#' @param escape_backslash same as in \code{readr::read_delim} (default FALSE)
+#' @param col_names same as in \code{readr::read_delim} (default TRUE)
+#' @param col_types same as in \code{readr::read_delim} (default NULL)
+#' @param locale same as in \code{readr::read_delim} (default default_locale())
+#' @param na same as in \code{readr::read_delim} (default c("", "NA"))
+#' @param quoted_na same as in \code{readr::read_delim} (default TRUE)
+#' @param comment same as in \code{readr::read_delim} (default "")
+#' @param trim_ws same as in \code{readr::read_delim} (default FALSE)
+#' @param skip same as in \code{readr::read_delim} (default 0)
+#' @param escape_double same as in \code{readr::read_delim} (default FALSE)
+#' @param callback A callback function to call on each chunk.
+#' @param show_progress_bar show progress bar (default TRUE)
+#'
+#' @return a SQLite database
+#'
+#' @export
+#' @importFrom DBI dbConnect dbDisconnect
+#' @importFrom RSQLite SQLite dbWriteTable
+#' @importFrom readr read_delim
+#' @importFrom dplyr %>% select_if mutate_at
+#' @importFrom lubridate is.Date is.POSIXt
+csv_to_sqlite <- function(csv_file, sqlite_file, table_name,
+                          callback = append_to_sqlite, delim = ",",
+                          pre_process_size = 1000, chunk_size = 50000,
+                          quote = "\"", escape_backslash = FALSE,
+                          col_names = TRUE, col_types = NULL,
+                          locale = default_locale(), na = c("", "NA"),
+                          quoted_na = TRUE, comment = "",
+                          trim_ws = FALSE, skip = 0,
+                          escape_double = TRUE,
+                          show_progress_bar = TRUE) {
+    con <- dbConnect(SQLite(), dbname = sqlite_file)
+
+    # read an extract of the data to extract the colnames and types
+    # to figure out the date and the datetime columns
+    df <- read_delim(csv_file, delim = delim, quote = quote,
+                     escape_backslash = escape_backslash,
+                     col_names = col_names, col_types = col_types,
+                     locale = locale, na = na, quoted_na = quoted_na,
+                     comment = comment, trim_ws = trim_ws,
+                     skip = skip, escape_double = escape_double,
+                     n_max = pre_process_size)
```

Comment:
Would it not be more convenient to use the [three dot ellipsis](https://www.r-bloggers.com/r-three-dots-ellipsis/) to pass the *additional* arguments (e.g. `delim`, `col_names`, `quote` directly to `the read_delim` function? That would for example also not require function adaptations in this function when `read_delim` get changed more upstream. Moreover, would release the function from a lot of the arguments, making it clearer.

Reply:
Ah yes, interesting! I will test it and let you know.

Reply:
As promised, done ffe889806bd9c647b5ff83483af4bae426f71eba